common/Root.tsx

```diff
@@ -152,6 +153,17 @@ class RootClass extends Component<Props, State> {
   }
 }
 
+let previousURL = '';
+const PageVisitsAnalytics = withRouter(({ history, children }) => {
+  history.listen(() => {
+    if (previousURL !== window.location.href) {
+      AnalyticsService.instance.trackPageVisit(window.location.href);
+      previousURL = window.location.href;
+    }
+  });
```

Comment:
Side effects should not be called in function components (except hooks). Alternatively, I think this should be extracted into a function that's called in componentDidMount/constructor in other components.